vyper/builtin_functions/functions.py

```diff
@@ -262,6 +262,11 @@ def build_LLL(self, expr, args, kwargs, context):
                     "Invalid start / length values needs to be between 0 and 32.", expr,
                 )
             sub_typ_maxlen = 32
+        elif sub.location == "calldata":
+            # if we are slicing msg.data, the length should
+            # be a constant, since msg.data can be of dynamic length
+            # we can't use it's length as the maxlen
+            sub_typ_maxlen = length.value
```

Comment:
please add a sanity check in addition to or instead of the comment:
```suggestion
            assert isinstance(length.value, int) # sanity check
            sub_typ_maxlen = length.value
```